src/object/README.md

```diff
@@ -54,7 +54,44 @@ erasure code over one target and OC\_EC\_16P2GX for 16+2 over all the available
 targets. See the full list in [daos_obj_class.h](/src/include/daos_obj_class.h)
 for more information.
 
-The object class ID and number of groups are embedded in object ID.
+### Object class naming conventions
+
+DAOS object supports two data protection methods: replication(RP) and Erasure Code(EC).
+A set of replicated shard, or a set of data and parity shards belonging to the same parity group
+is called redundancy group. An object can be chunked into multiple redundancy groups, which spread
+across many storage targets, in order to achieve higher I/O concurrency for better performance and
+large capacity. Targets for placing shards of a same redundancy group are selected from different
+fault domains, the default fault domain is "engine", it can be set to other domain like "node" or
```

Comment:
just a note that 2.4 release will set default fault domain as node, i have a pending patch PR9631 for that.